### PR TITLE
REF/TST: collect reduction tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
Similar to how to have tests.arithmetic for tests where we expect EA/Index/Series/DataFrame behavior to be analogous, this starts collecting the tests for reduction operations.  This is motivated by #24293.

This does not yet collect tests from tests.frame.test_analytics or any of tests.arrays or tests.indexes.

Optionally we may want parts of test_nanops to be part of test_reductions, since those functions will also be in the check-these-are-equivalent pile.

Parametrizes some tests in test_base.

Moved tests are unchanged, except for non-context usages of `pytest.raises` are context-ified; some tests are parametrized.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Thoughts on using markers instead of (or in addition to) moving?

**Comment by jbrockmendel** [comment]
> Thoughts on using markers instead of (or in addition to) moving?

The idea is to parametrize in order to make sure we test that behavior matches across the classes.  I don't see how using a marker instead gets us closer to that.

File: pandas/tests/test_reductions.py [hunk 1]
```diff
@@ -0,0 +1,1011 @@
...
+
+            result = getattr(s, method)(min_count=1)
+            assert pd.isna(result)
+
+            # Skipna, default
+            result = getattr(s, method)(skipna=True)
+            result == unit
+
+            # skipna, explicit
+            result = getattr(s, method)(skipna=True, min_count=0)
```
Comment (jreback) [inline pandas/tests/test_reductions.py]: does it make sense to make this a subdir to start?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_reductions.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/test_reductions.py]: I'm hopeful this won't be a big enough file to need to be split.  Easier to split later than un-split.

Comment (jreback) [reply pandas/tests/test_reductions.py]: its already a pretty hefty file no?

Comment (jbrockmendel) [reply pandas/tests/test_reductions.py]: i guess.  will split.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
ok, would like to remove the classes and rather have separate files for each. can leave that for a separate pass.

**Pull request merged**